### PR TITLE
Chromium support for Linux

### PR DESCRIPTION
**Pull request by chris-l**
Yes, about #1 issue. Is just a small change, as far as I know, the process is always called that.
I have 2 pcs with linux and chromium and it works ok in both of them.

If the pc is executing chrome **and** chromium, it should kill the tabs of _both_ browsers.

### CODE REVIEW

**Comment by arthurvr** [comment]
> This is for Chrome. Chromium is not Chrome. I'll consider adding a flag for Chromium and Chrome Canary.

Notice _"a flag"_. This just kills both Chrome and Chromium tabs w/o a flag.

**Comment by chris-l** [comment]
Oh sorry! How about this?

**Comment by sindresorhus** [comment]
@arthurvr I'm still not sure on it. Killing the tabs in all "chrome" browsers (Chrome, Chrome Canary, Chromium) might be nice as users wouldn't have to type the flag or think about which ones they're using. Thoughts?

// @kevva

**Comment by arthurvr** [comment]
> @arthurvr I'm still not sure on it. Killing the tabs in all "chrome" browsers (Chrome, Chrome Canary, Chromium) might be nice as users wouldn't have to type the flag or think about which ones they're using. Thoughts?

I'd say Chrome and Chrome canary, but I don't see a reason to do it for chromium. Personally, if I use chromium, it's always on side of my main chrome browser and for development purposes, not for browsing.

**Comment by sindresorhus** [comment]
@arthurvr Lots of Linux users use it as their main browser. As for your use-case, that's what I use Chrome Canary for.

**Comment by chris-l** [comment]
Eh well, I just uplodaded a new commit b7734f7e0b62641d9226bce8d4b364ffadcde171. This one kills chromium only by using a flag.

**Comment by arthurvr** [comment]
@sindresorhus What I meant is actually developing like, on the Chromium repo, developing Chromium. Sometimes (not to say most times) you can't do that in regular canary.

> Lots of Linux users use it as their main browser.

Makes sense. Maybe make it a flag, but make the default `true`.

**Comment by chris-l** [comment]
oh you meant a flag for killing both, not one or another? The flag I implemented kills chrome OR chromium.
`kill-tabs --chromium` kills only chromium
`kill-tabs` kills only chrome

**Comment by arthurvr** [comment]
My preference would be to just have all 3 be optional as flags, where Stable and Canary are on by default, and Chromium off by default.

**Comment by chris-l** [comment]
Heh, well I use chromium for developing and browsing. I don't have chrome installed.

@arthurvr, you mean flags like the ones used by `configure`?
Like:

```
kill-tabs --enable-chromium --disable-canary # kills chrome (is on by default) and chromium, but no canary
```

IMHO, isn't that too complicated? I think is more simple just:

```
kill-tabs # kills only chrome
kill-tabs --chromium # kills only chromium
kill-tabs --canary # kills only canary
```

But I can do it using `enable`/`disable`. @arthurvr, @sindresorhus, what you think?

**Comment by sindresorhus** [comment]
Too be more clear about my suggestion:

```
$ kill-tabs # kills tabs in Chrome and Chrome Canary and Chromium
```

Seems like the easiest solution. Nobody have to care about which browser they use and having to supply the correct flag. ?

## 

@chris-l Just hold on pushing new commits until we've decided what to go for ;)

**Comment by chris-l** [comment]
@sindresorhus  Heh, sure. By the way, since this commits are in my fork only, I think there is no problem for you and the parent repo if I squash the commits I did so far to whatever the final choice is. (no point in having useless commits)

I like your suggestion. Actually is similar to what the first commit I pushed (4ba3acef4e49464cb6d50064a68b6b049e494aed) did, except checking for canary.

**Comment by sindresorhus** [comment]
> By the way, since this commits are in my fork only

I'm aware. Just want to save you from unnessecary work while we go back and forth ;)

**Comment by arthurvr** [comment]
> Seems like the easiest solution. Nobody have to care about which browser they use and having to supply the correct flag. ?

Well, the times I use Chromium or Canary, it's not to actually browse the web, but for development (of eighter Chromium or simply sites). But I think I'm alone doing it like that, so yeah, that seems alright. I'd like it if there was a flag to turn one of them off though. Something like

```
$ kill-tabs # kills everything in Stable, Canary and Chromium
$ kill-tabs --no-chromium # Kills everything in Canary and Stable
```

**Comment by sindresorhus** [comment]
@arthurvr Alright, lets go for that. @chris-l :arrow_up:

**Comment by chris-l** [comment]
Ok, how about this commit?

Is missing the flag for canary, since I don't have canary and is not available for linux, but it has the flag for chrome and chromium. (adding the flag for canary would be as easy as adding a property to the `processes` object)

I installed and executed chrome along side chromium and tested that the flags actually work.

File: cli.js [hunk 1]
```diff
@@ -3,14 +3,17 @@
...
 var killTabs = require('./');
 
-meow({
+var cli = meow({
 	help: [
-		'Usage',
-		'  $ kill-tabs'
+		'Usage: kill-tabs [OPTIONS]',
+		'  --no-chromium       Do not kill tabs from chromium',
+		'  --chromium          Kill tabs from chromium (default)',
```
Comment (sindresorhus) [inline cli.js]: I think the `--no-` flags are enough. No point in this.

File: cli.js [hunk 2]
```diff
@@ -3,14 +3,17 @@
...
+		'  --no-chrome         Do not kill tabs from chrome',
+		'  --chrome            Kill tabs from chrome (default)'
```
Comment (sindresorhus) [inline cli.js]: I think the `--no-` flags are enough. No point in this.

File: cli.js [hunk 3]
```diff
@@ -3,14 +3,17 @@
...
 	]
 });
 
-killTabs(function (err) {
+killTabs(cli, function (err) {
```
Comment (sindresorhus) [inline cli.js]: You should pass in `cli.flags`, not the CLI. The API should have no knowledge of the CLI.

File: index.js [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
+  chrome : process.platform === 'darwin' ? 'Chrome Helper' : 'chrome',
+  chromium : 'chromium'
+};
 
 if (process.platform === 'win32') {
 	psList = require('./win');
 }
 
-module.exports = function (cb) {
+module.exports = function (cli, cb) {
```
Comment (sindresorhus) [inline index.js]: cli => opts

and should be optional

**Comment by sindresorhus** [comment]
This also needs to be done in https://github.com/sindresorhus/kill-tabs/blob/master/win.js

File: cli.js [hunk 5]
```diff
@@ -3,14 +3,17 @@
 var meow = require('meow');
 var killTabs = require('./');
 
-meow({
+var cli = meow({
 	help: [
-		'Usage',
-		'  $ kill-tabs'
+		'Usage: kill-tabs [OPTIONS]',
+		'  --no-chromium       Do not kill tabs from chromium',
```
Comment (arthurvr) [inline cli.js]: Do we also want such a flag for Canary, or is that technically impossible?

Comment (sindresorhus) [inline cli.js]: @arthurvr see https://github.com/sindresorhus/kill-tabs/pull/4#issuecomment-124905709 I can add that afterwards.

Comment (sindresorhus) [inline cli.js]: `Do not kill tabs from chromium` => `Do not kill tabs from Chromium`

File: cli.js [hunk 6]
```diff
@@ -3,14 +3,17 @@
...
+		'  --chromium          Kill tabs from chromium (default)',
+		'  --no-chrome         Do not kill tabs from chrome',
```
Comment (sindresorhus) [inline cli.js]: `Do not kill tabs from chrome` => `Do not kill tabs from Chrome`

File: cli.js (hunk 5 above)
Comment (arthurvr) [inline cli.js]: @sindresorhus Whoops, sorry, alright. :)

**Comment by chris-l** [comment]
Ok, it's done, except the part about `win.js`. Since that's related to windows, I think is better that someone else implements that part. (Not sure what is necessary to do there)

**Comment by kevva** [comment]
@sindresorhus, lgtm. I'll add support for Wndows as soon this is merged.

**Comment by sindresorhus** [comment]
Sure.

Note to self:
- Check what the Chromium helper process name is on OS X. I would assume it's `Chromium Helper`.
- Add support for Canary too.

**Pull request closed without merging**